torch/optim/lr_scheduler.py

```diff
@@ -23,6 +22,42 @@ def __init__(self, optimizer, last_epoch=-1):
         self.step(last_epoch + 1)
         self.last_epoch = last_epoch
 
+        self.state = kwargs
```

Comment:
This will easily get out of sync with attributes if someone changes them after the scheduler is created. It's best to avoid storing redundant information, and implement a specialized `__getstate__` for all schedulers

Reply:
Yes. Right. They would be out of sync, but we use the kwargs as a way to monitor what keys the state should contain. Our state_dict function then updates specifically only those keys.